libraries/support/azure/management.rb

```diff
@@ -141,22 +141,38 @@ def virtual_machine_disk(resource_group, id)
       )
     end
 
-    def subnet(resource_group, vnet, id)
+    def subnet(resource_group, vnet, name)
       get(
         url: link(location: "Microsoft.Network/virtualNetworks/#{vnet}/subnets",
-                    resource_group: resource_group) + id,
+                  resource_group: resource_group) + name,
         api_version: '2018-02-01',
       )
     end
 
     def subnets(resource_group, vnet)
       get(
         url: link(location: "Microsoft.Network/virtualNetworks/#{vnet}/subnets",
-                    resource_group: resource_group),
+                  resource_group: resource_group),
         api_version: '2018-02-01',
       )
     end
 
+    def sql_servers(resource_group)
+      get(
+        url: link(location: 'Microsoft.Sql/servers',
+                  resource_group: resource_group),
+        api_version: '2018-06-01-preview',
```

Comment:
Might be a good idea to have the `2018-06-01-preview` defined once  and injected here to make it easier to update in the future.  Could come from a configuration file or just a variable.

Reply:
Agreed. Each endpoint has a configured API version and by nature are not consistent. I think there is a future task to consolidate/parameterize the API versions we are using.